### ISSUE TITLE
Links wrapped in single quotes broken

### ISSUE THREAD

**Issue by mjmasn** [gregjacobs/Autolinker.js#265]
Links wrapped in single quotes incorrectly include the trailing quote in the url if the url contains one or more slashes...

:heavy_check_mark:  Go to "https://www.bbc.co.uk/news" (outputs Go to "[https://www.bbc.co.uk/news](https://www.bbc.co.uk/news)")

:heavy_check_mark:  Go to 'https://www.bbc.co.uk' (outputs Go to '[https://www.bbc.co.uk](https://www.bbc.co.uk)')

:boom: :cry:  Go to 'https://www.bbc.co.uk/news' (outputs Go to '[https://www.bbc.co.uk/news'](https://www.bbc.co.uk/news'))

**Comment by gregjacobs** [comment]
Hey, thanks for reporting! 

Interestingly, a single quote character is a valid path character. GitHub seems to exclude it if it's the last character of a URL though. Maybe I'll do the same.

Was also thinking to check the character before the URL to see if it's a single quote, but that may not cover the situation of `'go to https://www.bbc.co.uk/news'`

**Comment by mjmasn** [comment]
@gregjacobs ah yeah that makes sense... 

A 90% solution could be excluding a trailing single quote if it's the last character and doesn't have a matching single quote in the url.
```
'Go to https://www.bbc.co.uk/news?thing=this' <- exclude
https://www.bbc.co.uk/news?thing='hello' <- keep
(https://www.bbc.co.uk/news?thing=that's <- keep)
```
Seems fairly unlikely to have a url like:
```
https://www.bbc.co.uk/news?thing=that' or 
https://www.bbc.co.uk/news'
```

May have missed some scenarios though :thinking:

**Comment by gregjacobs** [comment]
@mjmasn Yes, I like your solution. I think that should cover most of the cases (or at least cover as many cases as we can cover!) 

Will do :)

**Comment by dbrgn** [comment]
The suggestion by @mjmasn sounds good 🙂 Still sesems to be an issue:

```js
> autolinker.link(`'see example.com/'`)
'see <a href="http://example.com/'"...>example.com/'</a>
```
